File: Test Case Generation/sonnet/cpp/test_task_19.cpp
#include <gtest/gtest.h>
#include <string>
#include <map>
using namespace std;

string sort_numbers(string numbers);

TEST(SortNumbersTest, EmptyString) {
    EXPECT_EQ(sort_numbers(""), "");
}

TEST(SortNumbersTest, SingleWordZero) {
    EXPECT_EQ(sort_numbers("zero"), "zero");
}

TEST(SortNumbersTest, SingleWordOne) {
    EXPECT_EQ(sort_numbers("one"), "one");
}

TEST(SortNumbersTest, SingleWordTwo) {
    EXPECT_EQ(sort_numbers("two"), "two");
}

TEST(SortNumbersTest, SingleWordThree) {
    EXPECT_EQ(sort_numbers("three"), "three");
}

TEST(SortNumbersTest, SingleWordFour) {
    EXPECT_EQ(sort_numbers("four"), "four");
}

TEST(SortNumbersTest, SingleWordFive) {
    EXPECT_EQ(sort_numbers("five"), "five");
}

TEST(SortNumbersTest, SingleWordSix) {
    EXPECT_EQ(sort_numbers("six"), "six");
}

TEST(SortNumbersTest, SingleWordSeven) {
    EXPECT_EQ(sort_numbers("seven"), "seven");
}

TEST(SortNumbersTest, SingleWordEight) {
    EXPECT_EQ(sort_numbers("eight"), "eight");
}

TEST(SortNumbersTest, SingleWordNine) {
    EXPECT_EQ(sort_numbers("nine"), "nine");
}

TEST(SortNumbersTest, AlreadySortedTwoWords) {
    EXPECT_EQ(sort_numbers("one five"), "one five");
}

TEST(SortNumbersTest, ReverseSortedTwoWords) {
    EXPECT_EQ(sort_numbers("five one"), "one five");
}

TEST(SortNumbersTest, ExampleFromDocstring) {
    EXPECT_EQ(sort_numbers("three one five"), "one three five");
}

TEST(SortNumbersTest, AllNumbersAscending) {
    EXPECT_EQ(sort_numbers("zero one two three four five six seven eight nine"),
              "zero one two three four five six seven eight nine");
}

TEST(SortNumbersTest, AllNumbersDescending) {
    EXPECT_EQ(sort_numbers("nine eight seven six five four three two one zero"),
              "zero one two three four five six seven eight nine");
}

TEST(SortNumbersTest, DuplicateNumbers) {
    EXPECT_EQ(sort_numbers("three three one"), "one three three");
}

TEST(SortNumbersTest, AllSameNumber) {
    EXPECT_EQ(sort_numbers("five five five"), "five five five");
}

TEST(SortNumbersTest, DuplicatesOfMultipleNumbers) {
    EXPECT_EQ(sort_numbers("two two one one zero zero"), "zero zero one one two two");
}

TEST(SortNumbersTest, ManyDuplicatesOfZero) {
    EXPECT_EQ(sort_numbers("zero zero zero"), "zero zero zero");
}

TEST(SortNumbersTest, ManyDuplicatesOfNine) {
    EXPECT_EQ(sort_numbers("nine nine nine"), "nine nine nine");
}

TEST(SortNumbersTest, TwoElementsEqual) {
    EXPECT_EQ(sort_numbers("six six"), "six six");
}

TEST(SortNumbersTest, MixedWithZeroAndNine) {
    EXPECT_EQ(sort_numbers("nine zero"), "zero nine");
}

TEST(SortNumbersTest, LargerUnsortedSet) {
    EXPECT_EQ(sort_numbers("six two four eight zero"), "zero two four six eight");
}

TEST(SortNumbersTest, RepeatedAllNumbers) {
    EXPECT_EQ(sort_numbers("nine zero eight one seven two six three five four"),
              "zero one two three four five six seven eight nine");
}

TEST(SortNumbersTest, OnlyZeros) {
    EXPECT_EQ(sort_numbers("zero zero zero zero zero"), "zero zero zero zero zero");
}

TEST(SortNumbersTest, OnlyNines) {
    EXPECT_EQ(sort_numbers("nine nine nine nine"), "nine nine nine nine");
}

TEST(SortNumbersTest, TwoExtremes) {
    EXPECT_EQ(sort_numbers("nine zero nine zero"), "zero zero nine nine");
}

TEST(SortNumbersTest, SingleZero) {
    EXPECT_EQ(sort_numbers("zero"), "zero");
}

TEST(SortNumbersTest, SingleNine) {
    EXPECT_EQ(sort_numbers("nine"), "nine");
}

TEST(SortNumbersTest, NoTrailingSpace) {
    string result = sort_numbers("one two three");
    EXPECT_NE(result.back(), ' ');
}

TEST(SortNumbersTest, NoLeadingSpace) {
    string result = sort_numbers("one two three");
    EXPECT_NE(result.front(), ' ');
}

TEST(SortNumbersTest, ResultHasCorrectWordCount) {
    string result = sort_numbers("three one five");
    int spaces = 0;
    for (char c : result) if (c == ' ') spaces++;
    EXPECT_EQ(spaces, 2);
}

TEST(SortNumbersTest, SortFiveElements) {
    EXPECT_EQ(sort_numbers("five four three two one"), "one two three four five");
}

TEST(SortNumbersTest, SortWithMiddleElement) {
    EXPECT_EQ(sort_numbers("nine five zero"), "zero five nine");
}

TEST(SortNumbersTest, AllOddWords) {
    EXPECT_EQ(sort_numbers("nine seven five three one"), "one three five seven nine");
}

TEST(SortNumbersTest, AllEvenWords) {
    EXPECT_EQ(sort_numbers("eight six four two zero"), "zero two four six eight");
}

TEST(SortNumbersTest, MixedDuplicatesAndUniques) {
    EXPECT_EQ(sort_numbers("one two one three two one"), "one one one two two three");
}

TEST(SortNumbersTest, TenDifferentNumbersRandomOrder) {
    EXPECT_EQ(sort_numbers("four seven one nine zero six three eight two five"),
              "zero one two three four five six seven eight nine");
}

int main(int argc, char **argv) {
    ::testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}

File: Test Case Generation/sonnet/python/test_task_19.py
import pytest
from solution import sort_numbers


def test_docstring_example():
    assert sort_numbers('three one five') == 'one three five'

def test_already_sorted():
    assert sort_numbers('zero one two three') == 'zero one two three'

def test_reverse_order():
    assert sort_numbers('nine eight seven six five four three two one zero') == 'zero one two three four five six seven eight nine'

def test_single_number_zero():
    assert sort_numbers('zero') == 'zero'

def test_single_number_nine():
    assert sort_numbers('nine') == 'nine'

def test_all_ten_numbers_unsorted():
    assert sort_numbers('five three nine zero two eight one seven four six') == 'zero one two three four five six seven eight nine'

def test_two_numbers_sorted():
    assert sort_numbers('one two') == 'one two'

def test_two_numbers_unsorted():
    assert sort_numbers('two one') == 'one two'

def test_duplicate_numbers():
    assert sort_numbers('three three three') == 'three three three'

def test_duplicates_mixed():
    assert sort_numbers('five one five two one') == 'one one two five five'

def test_all_same_number():
    assert sort_numbers('seven seven seven seven') == 'seven seven seven seven'

def test_empty_string():
    assert sort_numbers('') == ''

def test_returns_string_type():
    result = sort_numbers('one two three')
    assert isinstance(result, str)

def test_all_numbers_individually():
    words = ['zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine']
    for word in words:
        assert sort_numbers(word) == word

def test_zero_and_nine_only():
    assert sort_numbers('nine zero') == 'zero nine'

def test_four_five_six():
    assert sort_numbers('six four five') == 'four five six'

def test_multiple_duplicates_different_values():
    assert sort_numbers('two nine two nine') == 'two two nine nine'

def test_large_duplicate_set():
    assert sort_numbers('one one one one one') == 'one one one one one'

def test_sort_stability_preserves_order_for_equals():
    result = sort_numbers('three three')
    assert result == 'three three'

def test_eight_comes_before_nine():
    assert sort_numbers('nine eight') == 'eight nine'

def test_zero_smallest():
    assert sort_numbers('one zero') == 'zero one'

def test_nine_largest():
    assert sort_numbers('nine one') == 'one nine'